### PR TITLE
Initial rewrite of user guide module docs.

### PR DESCRIPTION
**Pull request by WRansohoff**
Currently only contains the `Flows` page.

I'll mark this as ready for review when the `PDKs`, `Checklists`, `Targets`, and possibly `Tools` pages are also ready.

### CODE REVIEW

File: docs/user_guide/flows.rst [hunk 1]
```diff
@@ -1,66 +1,73 @@
...
-     - sphinx
-     - yes
+SiliconCompiler flows are created by configuring the 'flowgraph' parameters within the schema. To simplify reuse of complex flows, the project includes standardized interfaces for bundling flowgraph settings as reusable named modules.
 
+Similar to other types of SiliconCompiler modules, flows are loaded by passing a :class:`.Flow` object into the :meth:`.use()` function before a run is started.
 
 setup(chip)
 -----------------
 
 A SiliconCompiler flowgraph consists of a set of connected nodes and edges, where a node is an executable tool performing some ("task"), and an edge is the connection between those tasks. The first task in the flowgraph must be named 'import'. ::
```
Comment (gadfort) [inline docs/user_guide/flows.rst:11]: Add a link to .node()

File: docs/user_guide/flows.rst [hunk 2]
```diff
@@ -1,66 +1,73 @@
...
 
-  chip.node(flow, 'import', <import_tool>)
-  chip.node(flow, <step>, <step_tool>)
-  chip.edge(flow, 'import', <step>)
+  flow.node(flow, 'import', <import_tool>)
+  flow.node(flow, <step>, <step_tool>)
+  flow.edge(flow, 'import', <step>)
```
Comment (gadfort) [inline docs/user_guide/flows.rst:15]: the signature of .node is <flow>, <step>, <tool>, <task>, so these should be updated to reflect that

File: docs/user_guide/pdks.rst [hunk 3]
```diff
@@ -45,33 +20,53 @@ An example of some of the fundamental settings are shown below.
...
     chip.set('pdk', process, 'version', <version>)
     chip.set('pdk', process, 'stackup', <stackuplist>)
-    chip.set('pdk', process, 'drc', <tool>, <stackup>, 'runset', <file>)
-    chip.set('pdk', process, 'lvs', <tool>, <stackup>, 'runset', <file>)
-    chip.set('pdk', process, 'devmodel', <stackup>, <modeltype>, <tool>, <file>)
+    chip.set('pdk', process, 'drc', 'runset', <tool>, <stackup>, <runset_type>, <file>)
+    chip.set('pdk', process, 'lvs', 'runset', <tool>, <stackup>, <runset_type> <file>)
+    chip.set('pdk', process, 'devmodel', <tool>, <modeltype>, <stackup>, <file>)
 
 To support standard RTL2GDS flows, the PDK setup will also need to specify pointers to routing technology rules, layout abstractions, layer maps, and routing grids as shown in the below example. For a complete set of available PDK parameters, see the :ref:`Schema<SiliconCompiler Schema>`. ::
```
Comment (gadfort) [inline docs/user_guide/pdks.rst:27]: make that a :keypath:`pdk` instead of generic link to schema

File: docs/user_guide/targets.rst [hunk 4]
```diff
@@ -1,38 +1,20 @@
 Targets
 ===================================
 
-To facilitate encapsulation and reuse of schema parameters related to design targets, SiliconCompiler implements a :meth:`.load_target()` function that dynamically loads Python modules located in special places on the filesystem.
+To facilitate encapsulation and reuse of schema parameters related to design targets, SiliconCompiler implements a :meth:`.load_target()` function which can run scripts that set up common combinations of :class:`.Flow`, :class:`.PDK`, :class:`.Library`, and :class:`.Checklist` modules.
 
-:meth:`.load_target()` takes in a string ``targetname``, and it will search for the path ``targets/<targetname>.py`` in the following locations, in this order:
+The :meth:`.load_target()` function takes in a string ``targetname``, and it will search for the path ``targets/<targetname>.py`` in the following locations, in this order:
```
Comment (gadfort) [inline docs/user_guide/targets.rst:6]: The search is actually for targets.<targetname>

File: docs/user_guide/targets.rst [hunk 5]
```diff
@@ -1,38 +1,20 @@
...
 
   #. The working directory from where the CLI app was called or the :class:`.Chip()` object instantiated.
   #. Paths specified in the :keypath:`option, scpath` schema parameter, separated by the OS-specific path separator (``:`` on Linux/macOS, ``;`` on Windows).
   #. Paths specified in the $SCPATH environment variable, separated by the OS-specific path separator.
   #. The root of the SiliconCompiler Python package, wherever it is installed.
```
Comment (gadfort) [inline docs/user_guide/targets.rst:11]: This no longer searches these paths

File: docs/user_guide/targets.rst [hunk 6]
```diff
@@ -1,38 +1,20 @@
...
 
-The ability to configure the search paths via a schema parameter or environment variable enables users to create custom targets and place them anywhere on their filesystem. Note that this file resolution scheme is also used by SC for resolving all other relative paths, not just target modules.
+The ability to configure the search paths via a schema parameter or environment variable enables users to create custom targets and place them anywhere on their filesystem. Most other types of modules are imported as Python libraries, which SiliconCompiler will look for in the system's Python search path.
```
Comment (gadfort) [inline docs/user_guide/targets.rst:13]: Now it's just python path

File: docs/user_guide/flows.rst [hunk 7]
```diff
@@ -1,66 +1,73 @@
...
-     - load_flow()
-     - yes
-
-   * - **make_docs**
-     - Doc generator
-     - None
-     - chip
-     - sphinx
-     - yes
+SiliconCompiler flows are created by configuring the 'flowgraph' parameters within the schema. To simplify reuse of complex flows, the project includes standardized interfaces for bundling flowgraph settings as reusable named modules.
```
Comment (gadfort) [inline docs/user_guide/flows.rst:4]: add link to flowgraph as :keypath:`flowgraph`

**Review by gadfort — CHANGES_REQUESTED** [review]
Just a quick skim of whats here.

File: docs/user_guide/pdks.rst [hunk 8]
```diff
@@ -24,7 +24,7 @@ An example of some of the fundamental settings are shown below.
     chip.set('pdk', process, 'lvs', 'runset', <tool>, <stackup>, <runset_type> <file>)
     chip.set('pdk', process, 'devmodel', <tool>, <modeltype>, <stackup>, <file>)
 
-To support standard RTL2GDS flows, the PDK setup will also need to specify pointers to routing technology rules, layout abstractions, layer maps, and routing grids as shown in the below example. For a complete set of available PDK parameters, see the :ref:`Schema<SiliconCompiler Schema>`. ::
+To support standard RTL2GDS flows, the PDK setup will also need to specify pointers to routing technology rules, layout abstractions, layer maps, and routing grids as shown in the below example. For a complete set of available PDK parameters, see the :keypath:`pdk` section of the :class:`.Schema`. ::
```
Comment (gadfort) [inline docs/user_guide/pdks.rst:27]: :Class:`schema` does nowhere, I think a :ref: is okay for that one

File: docs/user_guide/tools.rst [hunk 9]
```diff
@@ -264,15 +201,89 @@ It is the responsibility of the tool reference flow developer to bind the standa
...
+     - n/a
+     - run()
+     - no
+
+   * - **post_process**
+     - Post-executable logic
+     - :class:`.Chip`
+     - n/a
+     - run()
+     - yes
```
Comment (gadfort) [inline docs/user_guide/tools.rst:272]: no

File: docs/user_guide/tools.rst [hunk 10]
```diff
@@ -264,15 +201,89 @@ It is the responsibility of the tool reference flow developer to bind the standa
...
+     - list
+     - run()
+     - no
+
+   * - **parse_version**
+     - Returns executable version
+     - stdout
+     - version
+     - run()
+     - yes
```
Comment (gadfort) [inline docs/user_guide/tools.rst:251]: no

File: docs/user_guide/tools.rst [hunk 11]
```diff
@@ -264,15 +201,89 @@ It is the responsibility of the tool reference flow developer to bind the standa
...
+     - Description
+     - Arg
+     - Returns
+     - Used by
+     - Required
+
+   * - **setup**
+     - Configures tool
+     - :class:`.Chip`
+     - :class:`.Chip`
```
Comment (gadfort) [inline docs/user_guide/tools.rst:235]: does not return anything

**Review by gadfort — COMMENTED** [review]
I'm getting 
docs/user_guide/targets.rst:10: WARNING: undefined label: '.use()'
docs/user_guide/tools.rst:20: WARNING: undefined label: '.flow'

File: docs/user_guide/pdks.rst (hunk 8 above)
Comment (WRansohoff) [reply docs/user_guide/pdks.rst:27]: Looks like:

    :ref:`Schema <SiliconCompiler Schema>`

**Review by gadfort — APPROVED** [review]

**Pull request merged**